openedx/features/caliper_tracking/transformers/textbook_transformers.py

```diff
@@ -0,0 +1,48 @@
+"""
+Transformers for all the textbook interaction events
+"""
+import json
+
+
+def textbook_pdf_page_navigated(current_event, caliper_event):
+    """
+    The browser emits textbook.pdf.page.navigated events when a
+    user manually enters a page number.
+
+    :param current_event: default event log generated.
+    :param caliper_event: caliper_event log having some basic attributes.
+    :return: updated caliper_event.
+    """
+
+    caliper_event['actor'].update({
+        'type': 'Person',
+        'name': current_event['username']
+    })
+
+    caliper_event['referrer']['type'] = 'WebPage'
+
+    caliper_event['extensions']['extra_fields'].update({
+        'course_id': current_event['context'].get('course_id'),
+        'ip': current_event['ip'],
+    })
+
+    # "event" field of current_event is a string so convert it into object
```

Comment:
Never push comments to github. Also single line comments like this are not recommended in Python.